### DISCUSSION TITLE
How can I deploy an existing (local) Oqtane site to Azure?

### DISCUSSION THREAD

**Discussion by patrickjlee** [oqtane/oqtane.framework#4903] [Q&A]
I've seen the very detailed and helpful tutorial on how to deploy an empty Oqtane site to Azure at 
 https://blazorhelpwebsite.com/ViewBlogPost/1.

How can I deploy a site which already has content (i.e. pages, modules and themes) , developed locally (on a Windows pc) to Azure please?

**Comment by zyhfish** [comment]
This will be simple with below steps:
1. upload the website folder to azure app service via FTP
2. export the database from local sql server and restore to the azure sql server, you may need to upload the database export file into a storage account first;
3. update the connection string in config file to connect to the azure sql database, you may also need to update the records in alias table to use new domain of the azure app service.

**Reply by patrickjlee** [reply]
Thank you very much for the response!
Re 2 and 3. My database is already running in Azure, so there is nothing to migrate there, although I suppose I will need to add an alias for the site to allow for the azure url as well as the localhost one.

Re 1, when you say upload the website folder to azure app service, which folder do you mean?  And assuming the folder is called X, presumably I should upload its contents to the wwwroot folder of the azure app service, rather than putting the X folder as a subfolder within wwwroot?

**Reply by zyhfish** [reply]
yes, it's content but not the folder itself.

**Reply by patrickjlee** [reply]
Thanks- please may I have an answer to "when you say upload the website folder to azure app service, which folder do you mean"?

**Reply by zyhfish** [reply]
the folder which contains the oqtane application files:
![image](https://github.com/user-attachments/assets/c8c0eb4b-e6b2-45cc-96b6-2c71fdd70ae5)

**Reply by patrickjlee** [reply]
Sorry, but I can't find any folder with content like that, where is that within the local deployment? I've searched for a BlazorDebugProxy folder but can only find that within say Oqtane.Server/bin/Debug/net9.0/ but although that contains a BlazorDebugProxy folder, there is no wwwroot folder or Data or Packages folder.

Here are my top level folders:

![image](https://github.com/user-attachments/assets/9b251d4d-7e4a-446d-a5d4-ddbebce5d56b)

**Reply by sbwalker** [reply]
@patrickjlee if you are running Oqtane in a local development environment, the Oqtane.Server folder maps to the root folder of your deployed Azure environment. Your local environment has the following folders under Oqtane.Server:

```
/Content
  /Tenants <-- you should copy this folder (and subfolders and files)

/wwwroot
  /Content
    /Tenants <-- you should copy this folder (and subfolders and files)
```

**Reply by patrickjlee** [reply]
@sbwalker many thanks! I shall try that!
